### ISSUE TITLE
[Rule change] `avoid-non-null-assertion`: Shall we ignore cases like `SomeInheritedWidget.of(context)!`

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#719]
**What rule do you want to change?**
avoid-non-null-assertion

**How will the change be implemented? (New option, new default behavior, etc.)?**
See title. Will fill out more details if interested.

**Please provide some example code that this change will affect:**

```dart
```

**What does the rule currently do for this code?**

**What will the rule do after it's changed?**

**Are you willing to submit a pull request to implement this change?**

**Comment by incendial** [comment]
Is it a false-positive or there are some cases when it can be null?

**Comment by fzyzcjy** [comment]
The plugin is correct. But I guess this is how we commonly write down code(just like `map[key]!`)?

**Comment by incendial** [comment]
Been thinking about that for a while, maybe it's okay to leave as is? I'm really afraid that someone could get a null ref error in some case.

**Comment by fzyzcjy** [comment]
All right
